source/progress.ts

```diff
@@ -71,6 +72,19 @@ export function uploadProgress(request: ClientRequest, emitter: EventEmitter, up
 		const onSocketConnect = (): void => {
 			progressInterval = setInterval(() => {
 				const lastUploadedBytes = uploadedBytes;
+
+				/* istanbul ignore next: future versions of Node may not have this property */
+				if (!is.string((request as any)._header)) {
+					clearInterval(progressInterval);
+
+					const url = new URL('https://github.com/sindresorhus/got/issues/new');
+					url.searchParams.set('title', '`request._header` is not present');
```

Comment:
Later, we should open a Node.js issue about getting something official for this instead of having to use a private property.

Reply:
Will add a TODO here

Reply:
I'll do it now :)

Reply:
https://github.com/nodejs/node/issues/30496